util/src/main/java/com/google/cloud/hadoop/util/authorization/AuthorizationHandler.java

```diff
@@ -0,0 +1,123 @@
+package com.google.cloud.hadoop.util.authorization;
+
+import java.net.URI;
+import java.nio.file.AccessDeniedException;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * All custom authorization handler implementation should implement this interface.
+ *
+ * <p>The class is created by reflection. All customized AuthorizationHandler should implement this
+ * interface and read arguments through hadoop's Configurable interface.
+ */
+public interface AuthorizationHandler {
+
+  /**
+   * This method is called after instantiation and before calling handle. All properties with the
+   * prefix "fs.gs.authorization.properties." will be set to the properties map.
+   *
+   * @param properties Immutable map of properties.
+   */
+  void setConf(Map<String, String> properties);
+
+  /**
+   * Handles list object request.
```

Comment:
You may consider linking GCS API docs in javadoc for each method.

Example: https://cloud.google.com/storage/docs/json_api/v1/objects/list